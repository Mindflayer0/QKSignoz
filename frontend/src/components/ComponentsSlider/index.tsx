import React from 'react';

import menuItems from './menuItems';
import { Card, Container, Text } from './styles';

const DashboardGraphSlider = (): JSX.Element => {
	return (
		<Container>
			{menuItems.map(({ name, Icon }) => (
				<Card key={name} draggable>
					<Icon />
					<Text>{name}</Text>
				</Card>
			))}
		</Container>
	);
};

export default DashboardGraphSlider;
