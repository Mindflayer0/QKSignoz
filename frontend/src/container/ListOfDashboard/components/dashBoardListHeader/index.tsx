import { PlusOutlined } from '@ant-design/icons';
import { Dropdown, MenuProps, Row, Typography } from 'antd';
import TextToolTip from 'components/TextToolTip';
import { NewDashboardState } from 'container/ListOfDashboard';
import { useMemo } from 'react';
import { useTranslation } from 'react-i18next';

import { ButtonContainer, NewDashboardButton } from '../../styles';
import { getDashBoardText, getMenuItems } from './helpers';

function DashBoardListHeader({
	isDashboardListLoading,
	newDashboard,
	newDashboardState,
	createNewDashboard,
	onModalHandler,
	onNewDashboardHandler,
}: Props): JSX.Element {
	const { t } = useTranslation('dashboard');

	const menu: MenuProps = useMemo(
		() => ({
			items: getMenuItems({
				createNewDashboard,
				isDashboardListLoading,
				onModalHandler,
				onNewDashboardHandler,
				t,
			}),
		}),
		[
			createNewDashboard,
			isDashboardListLoading,
			onModalHandler,
			onNewDashboardHandler,
			t,
		],
	);

	return (
		<Row justify="space-between">
			<Typography>Dashboard List</Typography>

			<ButtonContainer>
				<TextToolTip
					text="More details on how to create dashboards"
					url="https:signoz.io/docs/userguide/dashboards"
				/>
				{newDashboard && (
					<Dropdown
						disabled={isDashboardListLoading}
						trigger={['click']}
						menu={menu}
					>
						<NewDashboardButton
							icon={<PlusOutlined />}
							type="primary"
							loading={newDashboardState.loading}
							danger={newDashboardState.error}
						>
							{getDashBoardText(newDashboardState)}
						</NewDashboardButton>
					</Dropdown>
				)}
			</ButtonContainer>
		</Row>
	);
}

interface Props {
	newDashboard: boolean;
	isDashboardListLoading: boolean;
	newDashboardState: NewDashboardState;
	createNewDashboard: boolean;
	onNewDashboardHandler: () => Promise<void>;
	onModalHandler: (uploadedGrafana: boolean) => void;
}

export default DashBoardListHeader;
