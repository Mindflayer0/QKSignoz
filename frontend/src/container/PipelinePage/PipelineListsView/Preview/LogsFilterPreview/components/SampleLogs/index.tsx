import './styles.scss';

import { Time } from 'container/TopNav/DateTimeSelection/config';
import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';

import LogsList from '../../../components/LogsList';
import useSampleLogs from '../../../hooks/useSampleLogs';

function SampleLogs({ filter, timeInterval }: SampleLogsProps): JSX.Element {
	const { isLoading, isError, logs } = useSampleLogs({
		filter,
		timeInterval,
		count: 5,
	});

	if (isError) {
		return (
			<div className="sample-logs-notice-container">
				could not fetch logs for filter
			</div>
		);
	}

	if (isLoading) {
		return <div className="sample-logs-notice-container">Loading...</div>;
	}

	if ((filter?.items?.length || 0) < 1) {
		return (
			<div className="sample-logs-notice-container">Please select a filter</div>
		);
	}

	if (logs.length < 1) {
		return <div className="sample-logs-notice-container">No logs found</div>;
	}

	return <LogsList logs={logs} />;
}

interface SampleLogsProps {
	filter: TagFilter;
	timeInterval: Time;
}

export default SampleLogs;
