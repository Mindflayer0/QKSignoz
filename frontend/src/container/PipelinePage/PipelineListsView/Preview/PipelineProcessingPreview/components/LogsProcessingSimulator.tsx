import { Button } from 'antd';
import { useState } from 'react';
import { ILog } from 'types/api/logs/log';
import { PipelineData } from 'types/api/pipeline/def';

function LogsProcessingSimulator({
	inputLogs,
	pipeline,
}: LogsProcessingSimulatorProps): JSX.Element {
	const [showResult, setShowResult] = useState<boolean>(false);
	const simulate = (): void => setShowResult(true);

	if (!showResult) {
		return (
			<div>
				<Button type="primary" onClick={simulate}>
					Simulate Processing
				</Button>
			</div>
		);
	}

	return <div>{JSON.stringify({ inputLogs, pipeline })}</div>;
}

export interface LogsProcessingSimulatorProps {
	inputLogs: ILog[];
	pipeline: PipelineData;
}

export default LogsProcessingSimulator;
