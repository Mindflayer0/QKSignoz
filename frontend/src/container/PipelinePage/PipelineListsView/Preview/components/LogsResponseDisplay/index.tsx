import { LogsResponse } from '../../types';
import LogsList from '../LogsList';

function LogsResponseDisplay({
	response,
}: LogsResponseDisplayProps): JSX.Element {
	const { isLoading, isError, logs } = response;

	if (isError) {
		return (
			<div className="sample-logs-notice-container">There was an error.</div>
		);
	}

	if (isLoading) {
		return <div className="sample-logs-notice-container">Loading...</div>;
	}

	if (logs.length < 1) {
		return <div className="sample-logs-notice-container">No logs found</div>;
	}

	return <LogsList logs={logs} />;
}

export interface LogsResponseDisplayProps {
	response: LogsResponse;
}

export default LogsResponseDisplay;
