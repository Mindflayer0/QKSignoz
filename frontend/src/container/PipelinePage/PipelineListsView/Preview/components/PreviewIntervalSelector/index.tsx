import './styles.scss';

import { Select } from 'antd';
import {
	initialFilters,
	initialQueriesMap,
	PANEL_TYPES,
} from 'constants/queryBuilder';
import {
	RelativeDurationOptions,
	Time,
} from 'container/TopNav/DateTimeSelection/config';
import { useGetQueryRange } from 'hooks/queryBuilder/useGetQueryRange';
import { cloneDeep } from 'lodash-es';
import { useMemo } from 'react';
import { TagFilter } from 'types/api/queryBuilder/queryBuilderData';
import { LogsAggregatorOperator } from 'types/common/queryBuilder';

function MatchedLogsCount({
	filter,
	timeInterval,
}: MatchedLogsCountProps): JSX.Element {
	const query = useMemo(() => {
		const q = cloneDeep(initialQueriesMap.logs);
		q.builder.queryData[0] = {
			...q.builder.queryData[0],
			filters: filter || initialFilters,
			aggregateOperator: LogsAggregatorOperator.COUNT,
		};
		return q;
	}, [filter]);

	const result = useGetQueryRange({
		graphType: PANEL_TYPES.TABLE,
		query,
		selectedTime: 'GLOBAL_TIME',
		globalSelectedInterval: timeInterval,
	});

	if (result.isFetched) {
		const count =
			result?.data?.payload?.data?.newResult?.data?.result?.[0]?.series?.[0]
				?.values?.[0]?.value;
		return (
			<div className="logs-filter-preview-matched-logs-count">
				{count} matches in
			</div>
		);
	}
	return <div />;
}

interface MatchedLogsCountProps {
	filter: TagFilter;
	timeInterval: Time;
}

function PreviewIntervalSelector({
	previewFilter,
	value,
	onChange,
}: PreviewIntervalSelectorProps): JSX.Element {
	const isEmptyFilter = (previewFilter?.items?.length || 0) < 1;

	return (
		<div className="logs-filter-preview-time-interval-summary">
			{!isEmptyFilter && (
				<MatchedLogsCount filter={previewFilter} timeInterval={value} />
			)}
			<div>
				<Select
					onSelect={(value: unknown): void => onChange(value as Time)}
					value={value}
				>
					{RelativeDurationOptions.map(({ value, label }) => (
						<Select.Option key={value + label} value={value}>
							{label}
						</Select.Option>
					))}
				</Select>
			</div>
		</div>
	);
}

interface PreviewIntervalSelectorProps {
	value: Time;
	onChange: (interval: Time) => void;
	previewFilter: TagFilter;
}

export default PreviewIntervalSelector;
