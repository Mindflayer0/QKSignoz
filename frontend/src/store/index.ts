import { applyMiddleware, compose, createStore } from 'redux';
import type { ThunkMiddleware } from 'redux-thunk';
import thunk from 'redux-thunk';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	// @TODO Add Type for AppActions also
	composeEnhancers(applyMiddleware(thunk as ThunkMiddleware<AppState>)),
);

export default store;
