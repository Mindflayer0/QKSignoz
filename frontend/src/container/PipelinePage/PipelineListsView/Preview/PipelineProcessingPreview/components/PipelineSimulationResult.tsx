import { ILog } from 'types/api/logs/log';
import { PipelineData } from 'types/api/pipeline/def';

import LogsResponseDisplay from '../../components/LogsResponseDisplay';
import usePipelinePreview from '../../hooks/usePipelinePreview';

function PipelineSimulationResult({
	inputLogs,
	pipeline,
}: PipelineSimulationResultProps): JSX.Element {
	const simulationResult = usePipelinePreview({ pipeline, logs: inputLogs });

	return <LogsResponseDisplay response={simulationResult} />;
}

export interface PipelineSimulationResultProps {
	inputLogs: ILog[];
	pipeline: PipelineData;
}

export default PipelineSimulationResult;
