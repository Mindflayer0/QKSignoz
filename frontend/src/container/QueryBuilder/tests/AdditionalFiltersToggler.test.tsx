import { render } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import i18n from 'ReactI18';
import store from 'store';

import { AdditionalFiltersToggler } from '../components';

describe('QueryBuilder: Toggler for Additional Filters', () => {
	it('it should render AdditionalFiltersToggler', () => {
		const { asFragment } = render(
			<MemoryRouter>
				<Provider store={store}>
					<I18nextProvider i18n={i18n}>
						<AdditionalFiltersToggler listOfAdditionalFilter={[]}>
							Testing AdditionalFiltersToggler
						</AdditionalFiltersToggler>
					</I18nextProvider>
				</Provider>
			</MemoryRouter>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
