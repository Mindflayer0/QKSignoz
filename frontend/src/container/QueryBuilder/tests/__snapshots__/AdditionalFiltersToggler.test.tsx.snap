// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryBuilder: Toggler for Additional Filters it should render AdditionalFiltersToggler 1`] = `
<DocumentFragment>
  .c1 {
  margin-right: 0.6875rem;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.875rem;
  min-height: 1.375rem;
  cursor: pointer;
}

<div
    class="ant-row css-dev-only-do-not-override-1i536d8"
  >
    <div
      class="ant-col ant-col-24 css-dev-only-do-not-override-1i536d8"
    >
      <div
        class="ant-col c0 css-dev-only-do-not-override-1i536d8"
      >
        <span
          aria-label="plus-square"
          class="anticon anticon-plus-square c1"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus-square"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
            />
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
            />
          </svg>
        </span>
        <span>
          Add conditions for 
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
